code/scripting/scripting.cpp

```diff
@@ -510,42 +540,29 @@ bool ConditionedHook::ConditionsValid(int action, object *objp, int more_data)
 
 					int action_index = more_data;
 
-					if (action_index <= 0 || stricmp(scp->condition_string.c_str(), Control_config[action_index].text.c_str()) != 0)
+					if (action_index <= 0 || stricmp(scp.condition_string.c_str(), Control_config[action_index].text.c_str()) != 0)
 						return false;
 					break;
 				}
 			case CHC_VERSION:
 				{
 					// Goober5000: I'm going to assume scripting doesn't care about SVN revision
-					char buf[32];
-					sprintf(buf, "%i.%i.%i", FS_VERSION_MAJOR, FS_VERSION_MINOR, FS_VERSION_BUILD);
-					if(stricmp(buf, scp->condition_string.c_str()) != 0)
+					
+					//if(stricmp(Script_engine_version.c_str(), scp.condition_string.c_str()) != 0)
+					//This is pre-checked and the result cached on script load, since version string 
+					//doesn't change during play and building the version string over and over to compare to
+					//was more expensive than you might think
+					if(scp.condition_reference_index == 0)
 					{
-						//In case some people are lazy and say "3.7" instead of "3.7.0" or something
-						if(FS_VERSION_BUILD == 0)
-						{
-							sprintf(buf, "%i.%i", FS_VERSION_MAJOR, FS_VERSION_MINOR);
-							if(stricmp(buf, scp->condition_string.c_str()) != 0)
-								return false;
-						}
-						else
-						{
-							return false;
-						}
+						return false;
 					}
 					break;
 				}
 			case CHC_APPLICATION:
 				{
-					if(Fred_running)
+					if(scp.condition_reference_index == 0)
```

Comment:
Please also add a comment here what the value of `condition_reference_index` means in this case.